File: src/lib/plate.ts
/** Mask plate for privacy: 浙A12348 → 浙A···8 */
export function maskPlate(plate: string): string {
  const p = plate.trim().toUpperCase()
  if (!p) return '···'
  if (p.length <= 3) return `${p[0] ?? ''}···`
  return `${p.slice(0, 2)}···${p.slice(-1)}`
}

File: server/app/plate.py
"""Plate masking helpers (Chinese-plate friendly)."""

from __future__ import annotations


def mask_plate(plate: str) -> str:
    """Mask plate for privacy: 浙A12348 → 浙A···8"""
    p = plate.strip().upper()
    if not p:
        return "···"
    if len(p) <= 3:
        return f"{p[0] if p else ''}···"
    return f"{p[:2]}···{p[-1]}"
